src/commcare_cloud/fab/operations/release.py

```diff
@@ -68,8 +68,6 @@ def _update_code_from_previous_release(code_repo, subdir, git_env):
         code_root = os.path.join(code_root, subdir)
 
     if files.exists(code_current, use_sudo=True):
-        with cd(code_current), shell_env(**git_env):
-            sudo('git submodule foreach "git fetch origin"')
```

Comment:
This removal is the one concern I have. This fetch appears to be unnecessary and grabs all branches, but it was added in a standalone PR (https://github.com/dimagi/commcare-cloud/pull/2787) so ostensibly was necessary at one point. However, it should never be necessary to fetch in the submodule since `git submodule update` handles that.

Reply:
Agreed. This is what `submodule update` does for you. Furthermore, performing a `foreach 'git fetch origin'` will do exactly what this PR intends to stop (indiscriminately fetch _all_ remote branches), so it simply needs to go, one way or another.